src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -1654,7 +1654,7 @@ trait Namers extends MethodSynthesis {
         if (sym hasAnnotation NativeAttr)
           sym resetFlag DEFERRED
         else {
-          if (!symbolAllowsDeferred && ownerRequiresConcrete) fail(AbstractVar)
+          if (!symbolAllowsDeferred && ownerRequiresConcrete && !sym.isJava) fail(AbstractVar)
```

Comment:
I would expected `ownerRequiresConcrete` to be `false` because the methods owner is a class but not a module class nor an anonymous class.